### PR TITLE
Added haplotag metrics

### PR DESCRIPTION
**Pull request by jenniferliddle**

### CODE REVIEW

File: lib/npg_qc/autoqc/checks/haplotag_metrics.pm [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+  $self->result->{'clear_file'} = $clear_file;
+  $self->result->{'unclear_file'} = $unclear_file;
+  $self->result->{'missing_file'} = $missing_file;
+
+  $self->result->{'clear_count'} = $clear_count;
+  $self->result->{'unclear_count'} = $unclear_count;
+  $self->result->{'missing_count'} = $missing_count;
+
+  my $pass = ($clear_count + $unclear_count) / ($clear_count + $unclear_count + $missing_count);
```
Comment (dkj) [inline lib/npg_qc/autoqc/checks/haplotag_metrics.pm:90]: Is this always 100% for our data?

Comment (jenniferliddle) [reply lib/npg_qc/autoqc/checks/haplotag_metrics.pm:90]: The SamHaplotag program puts reads with no BX tag assigned into the 'Missing' file, and all others into the 'Clear' and 'Unclear' files.
We may want to change the $PASS_RATE from 0.8 to 0.9 (my notes (ie handwriting) is a little unclear on that.

Comment (dkj) [reply lib/npg_qc/autoqc/checks/haplotag_metrics.pm:90]: Does 'Missing' ever have any data for our pipelines though? I think all our data has BC auxtag to do the calculation on. I _suspect_ dud haplotags get counted in 'Unclear'.

N.b. @dozy a good example to check would be where we've done the "revcomp" incorrectly - that shoudl come out as a clear fail.

Comment (jenniferliddle) [reply lib/npg_qc/autoqc/checks/haplotag_metrics.pm:90]: I'm not clear what 'dud haplotags' means in this context. The test was supposed to be 'N% of reads assigned a BX tag'. We expect most of our data to pass this test, but wanted to check for an occasional fail.

Comment (dkj) [reply lib/npg_qc/autoqc/checks/haplotag_metrics.pm:90]: Are we always getting BX assigned? If it's always 100% then that criteria isn't useful - misapplying the revcomp produces data which breaks downstream analysis, as such it is a good 1st test whether what we're doing (the pass criteria) has value.

**Pull request merged**